### PR TITLE
simplified and sped up metrics list rendering

### PR DESCRIPTION
**Pull request by BrandonArp**

### CODE REVIEW

File: app/assets/javascripts/classes/GraphViewModel.ts [hunk 1]
```diff
@@ -224,252 +230,12 @@ module GraphViewModel {
...
 
     export var createMetric = (spec: GraphSpec) => {
-        var currFolder = getRootFolderMetric(spec.service);
-        var metricSplit = spec.metric.split("/");
-        if (metricSplit.length > 1) {
-            addMetricFolder(spec.service, spec.metric, [spec.statistic], metricSplit, "", currFolder);
-        } else {
-            createFolderMetric(spec, metricSplit[0], currFolder);
-        }
+        //metricsList.addStatistic(spec);
```
Comment (vjkoskela) [inline app/assets/javascripts/classes/GraphViewModel.ts]: Should this be removed?

File: app/assets/javascripts/classes/MetricsBrowseList.ts [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+        }
+        value = value.replace(/ /g, "_").toLowerCase();
+        return value.replace(/\//g, "_");
+    };
+
+    private getPathParts(path: string): string[] {
+        return path.split("/");
+    };
+
+
```
Comment (vjkoskela) [inline app/assets/javascripts/classes/MetricsBrowseList.ts]: unnecessary blank line.

File: app/assets/javascripts/classes/MetricsBrowseList.ts [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    private idify(value: string): string {
+        if (value == undefined) {
+            return undefined;
+        }
+        value = value.replace(/ /g, "_").toLowerCase();
+        return value.replace(/\//g, "_");
+    };
+
+    private getPathParts(path: string): string[] {
```
Comment (vjkoskela) [inline app/assets/javascripts/classes/MetricsBrowseList.ts]: not sure what our js style ordering is here, but the methods seem to be in inconsistent order based on scope here.

File: app/assets/javascripts/classes/MetricsBrowseList.ts [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+        }
+        if (cmp == 0) {
+            return haystack[current];
+        } else {
+            return undefined;
+        }
+    }
+
+    private nodeComparator(first: BrowseNode, second: BrowseNode): number {
+        //if (first.constructor == second.constructor) {
```
Comment (vjkoskela) [inline app/assets/javascripts/classes/MetricsBrowseList.ts]: Can we remove the unused code?

File: app/assets/javascripts/classes/MetricsBrowseList.ts [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+        //} else if (first instanceof FolderNodeVM) {
+        //    return -1;
+        //} else if (second instanceof FolderNodeVM) {
+        //    return 1;
+        //}
+    }
+
+    private stringCompare(first: string, second: string): number {
+        return ns.naturalSort(first, second);
+        //return first == second ? 0 : first > second ? 1 : -1;
```
Comment (vjkoskela) [inline app/assets/javascripts/classes/MetricsBrowseList.ts]: Same.

File: app/assets/javascripts/metrics_portal.ts [hunk 6]
```diff
@@ -20,7 +20,9 @@
 import system = require('durandal/system');
 import app = require('durandal/app');
 import viewLocator = require('durandal/viewLocator');
+import ko = require('knockout');
 
+//ko.options.deferUpdates = true;
```
Comment (vjkoskela) [inline app/assets/javascripts/metrics_portal.ts]: ?

File: app/assets/javascripts/classes/GraphViewModel.ts (hunk 1 above)
Comment (BrandonArp) [inline app/assets/javascripts/classes/GraphViewModel.ts]: Done.

File: app/assets/javascripts/classes/MetricsBrowseList.ts (hunk 2 above)
Comment (BrandonArp) [inline app/assets/javascripts/classes/MetricsBrowseList.ts]: Fixed.

File: app/assets/javascripts/classes/MetricsBrowseList.ts (hunk 3 above)
Comment (BrandonArp) [inline app/assets/javascripts/classes/MetricsBrowseList.ts]: Fixed.

File: app/assets/javascripts/classes/MetricsBrowseList.ts (hunk 4 above)
Comment (BrandonArp) [inline app/assets/javascripts/classes/MetricsBrowseList.ts]: Fixed.

File: app/assets/javascripts/classes/MetricsBrowseList.ts (hunk 5 above)
Comment (BrandonArp) [inline app/assets/javascripts/classes/MetricsBrowseList.ts]: Fixed.

File: app/assets/javascripts/metrics_portal.ts (hunk 6 above)
Comment (BrandonArp) [inline app/assets/javascripts/metrics_portal.ts]: Fixed.

**Pull request merged**